python/res/simulator/simulation_context.py

```diff
@@ -19,52 +15,44 @@ def __init__(self, ert, sim_fs, mask, itr , verbose=False):
         job_queue = ert.get_queue_config().create_job_queue()
         job_queue.set_max_job_duration(max_runtime)
         self._queue_manager = JobQueueManager(job_queue)
-        self._queue_manager.startQueue( mask.count( ), verbose=verbose)
-        self._run_args = {}
-        """ :type: dict[int, RunArg] """
-
-        self._thread_pool = CThreadPool(8)
-        self._thread_pool.addTaskFunction("submitJob", RES_LIB, "enkf_main_isubmit_job__")
 
         subst_list = self._ert.getDataKW( )
         path_fmt = self._ert.getModelConfig().getRunpathFormat()
         jobname_fmt = self._ert.getModelConfig().getJobnameFormat()
-        self._run_context = ErtRunContext( EnkfRunType.ENSEMBLE_EXPERIMENT, sim_fs, None, mask, path_fmt, jobname_fmt, subst_list, itr)
-        self._ert.initRun(self._run_context)
 
+        self._run_context = ErtRunContext( EnkfRunType.ENSEMBLE_EXPERIMENT, sim_fs, None, mask, path_fmt, jobname_fmt, subst_list, itr)
+        self._ert.getEnkfSimulationRunner().createRunPath(self._run_context)
+        self.sim_thread = self._run_simulations_simple_step()
+
+    def _get_run_args(self, iens):
+        '''
+        raises an  exception if no iens simulation found
+
+        :param iens: realization number
+        :return: run_args for the realization
+        '''
+        for run_arg in self._run_context:
+            if run_arg is not None and run_arg.iens == iens:
+                return run_arg
+        raise KeyError("No such simulation: %s" % iens)
+
+    def _run_simulations_simple_step(self):
+        sim_thread = Thread(
+            target=lambda: self._ert.getEnkfSimulationRunner().runSimpleStep(self._queue_manager.get_job_queue(), self._run_context)
+        )
+        sim_thread.start()
+        return sim_thread
 
     def __len__(self):
         return self._mask.count()
 
-
-    def addSimulation(self, iens, geo_id):
-        if not (0 <= iens < len(self._run_context)):
-            raise UserWarning("Realization number out of range: %d >= %d" % (iens, len(self._run_context)))
-
-        if not self._mask[iens]:
-            raise UserWarning("Realization number: '%d' is not active" % iens)
-
-        if iens in self._run_args:
-            raise UserWarning("Realization number: '%d' already queued" % iens)
-
-        run_arg = self._run_context[iens]
-        run_arg.geo_id = geo_id
```

Comment:
This is crucial and should be preserved. See comment on `geo_id` being lost. Could be passed to the `BatchContext` upon creation.

Reply:
Note: We should ensure that the tests actually capture this behavior

Reply:
Agree, we can sit down and suggest a proper test for capturing the `geo_id`

Reply:
Decided just to do a simple `geo_id` assert in `test_simulation_context`